### PR TITLE
#11795: Update test_pgm_dispatch and sweep

### PR DESCRIPTION
**Pull request by pgkeller**
Add dispatch to eth tests
Add dispatch from eth flag
Add long running + shadow kernel tests
Add a combo of the above

### Ticket
Test only no ticket

### Problem description
Update dispatch sweep

### What's changed
See above

### Checklist
- [ x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp [hunk 1]
```diff
@@ -207,6 +217,29 @@ void initialize_program(tt_metal::Program& program, uint32_t run_cycles) {
         kg.start_coord = { kg.end_coord.x + 1, kg.end_coord.y };
         kg.end_coord = kg.start_coord;
     }
+
+    if (erisc_enabled_g) {
+        auto erisc_cores = device->get_active_ethernet_cores();
```
Comment (tt-aho) [inline tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp:222]: Should it be `get_active_ethernet_cores(true)`?

Comment (pgkeller) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp:222]: yes, I think so (just talked w/ David who debugged this bug in CI recently)

Comment (pgkeller) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp:222]: will fix

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp [hunk 2]
```diff
@@ -218,17 +251,18 @@ int main(int argc, char **argv) {
     try {
         int device_id = 0;
         tt_metal::Device* device;
+        DispatchCoreType dispatch_core_type = dispatch_from_eth_g ? DispatchCoreType::ETH : DispatchCoreType::WORKER;
         if (use_trace_g) {
-            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000);
+            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000, dispatch_core_type);
         } else {
-            device = tt_metal::CreateDevice(device_id);
+            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000, dispatch_core_type);
```
Comment (sagarwalTT) [inline tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp:258]: In this `if-else` block, `device` is being initialized the same way in both cases. Should it be different in each case?

Comment (pgkeller) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp:258]: good catch, these code paths converged, will fix

**Review by tt-aho — APPROVED** [review]

**Review by abhullar-tt — APPROVED** [review]

**Pull request merged**